### PR TITLE
EDO-84 animations when completing ksill

### PR DESCRIPTION
**Pull request by 5Heinzelmann**
If you complete a skill, you see a firework animation behind your avatar.
Also: If your score updates (completing but also loosing a skill) it gets a little animation too.

### CODE REVIEW

**Comment by fbelter-iteratec** [comment]
Shouldn't package-lock.json be committed?

File: src/main/webapp/app/teams/teams.scss [hunk 1]
```diff
@@ -18,3 +22,93 @@
...
+    @-ms-keyframes #{$animationName} {
+        @content;
+    }
+
+    @keyframes #{$animationName} {
+        @content;
+    }
+}
+
+@mixin animation-delay($settings) {
```
[the remark this answers is not in the document]
Comment (5Heinzelmann) [reply src/main/webapp/app/teams/teams.scss]: No, I will remove it.

File: src/main/webapp/app/teams/teams.scss [hunk 2]
```diff
@@ -18,3 +22,93 @@
...
+// Firework Animation
+$particles: 200;
+$width: 500;
+$height: 600;
+
+$end: ();
+$start: ();
+@for $i from 0 through $particles {
+    $end: $end, random($width)-$width / 2 + px random($height)-$height / 1.2 + px hsl(18, 100, random(90) + 5);
+    $start: $start, 0 0 #e8ddb7;
```
[the remark this answers is not in the document]
Comment (5Heinzelmann) [reply src/main/webapp/app/teams/teams.scss]: The color doesn't matter at this point. The animation starts behind the avatar, so you can't see the initial color value.

**Comment by 5Heinzelmann** [comment]
> Shouldn't package-lock.json be committed?

@ngnmhieu @sberlin ?

File: src/main/webapp/app/teams/teams-status.component.ts [hunk 3]
```diff
@@ -14,11 +14,14 @@ import { TeamsEditComponent } from 'app/teams/teams-edit.component';
...
 import { take } from 'rxjs/operators';
+import { trigger, transition, useAnimation } from '@angular/animations';
+import { flash } from 'ng-animate';
 
 @Component({
     selector: 'jhi-teams-status',
     templateUrl: './teams-status.component.html',
-    styleUrls: ['teams-status.scss']
+    styleUrls: ['teams-status.scss'],
+    animations: [trigger('score', [transition('* <=> *', useAnimation(flash))])]
```
Comment (sberlin) [inline src/main/webapp/app/teams/teams-status.component.ts]: Is the animation on page load intended?

File: src/main/webapp/app/teams/teams.component.html [hunk 4]
```diff
@@ -18,3 +18,4 @@
         </div>
     </div>
 </div>
+<div id="pyro" style="display: none;" class="pyro"></div>
```
Comment (sberlin) [inline src/main/webapp/app/teams/teams.component.html]: Move the style to the class.

File: src/main/webapp/app/teams/teams-status.component.ts (hunk 3 above)
Comment (5Heinzelmann) [reply src/main/webapp/app/teams/teams-status.component.ts]: No, do you have an idea how to fix that?

File: src/main/webapp/app/teams/teams.scss [hunk 5]
```diff
@@ -18,3 +22,77 @@
...
+
+.pyro {
+    position: absolute;
+    top: 25%;
+    left: 25%;
+    width: 6px;
+    height: 6px;
+    border-radius: 100%;
+    box-shadow: $start;
+    @include animation((1.2s bang ease-out, 1.2s gravity ease-in));
```
[the remark this answers is not in the document]
Comment (5Heinzelmann) [reply src/main/webapp/app/teams/teams.scss]: Increasing both 1.2s to 1.7s? 1.7 still laggs for me a little bit, just the whole animation is a little slower..

File: src/main/webapp/app/teams/teams.component.ts [hunk 6]
```diff
@@ -39,6 +37,12 @@ export class TeamsComponent implements OnInit {
     loadTeamSkills() {
         this.teamSkillService.query({ 'teamId.equals': this.team.id }).subscribe(teamSkillResponse => {
             this.team.skills = this.teamSkills = teamSkillResponse.body;
+            if (this.team.skills.find(skill => moment().diff(skill.completedAt, 'seconds') < 2)) {
```
[the remark this answers is not in the document]
Comment (5Heinzelmann) [reply src/main/webapp/app/teams/teams.component.ts]: I agree that binding the animation to a direct event would be better, but unfortunately I have no idea how to implement that properly. Do you have a suggestion? (Than we could get rid of the magic 2 seconds also)

**Comment by 5Heinzelmann** [comment]
I think we should discuss which kind of animation we want to show and make a new PR for that.

**Pull request closed without merging**